src/views/Package/index.tsx

```diff
@@ -0,0 +1,52 @@
+import { Box } from "@chakra-ui/react";
+import * as reflect from "jsii-reflect";
+import { useEffect } from "react";
+import { useParams, useLocation } from "react-router-dom";
+import { createAssembly, fetchMetadata, Metadata } from "../../api/package";
+import { PackageDetails } from "../../components/PackageDetails";
+import { PackageDocs } from "../../components/PackageDocs";
+import { useRequest } from "../../hooks/useRequest";
+import { parseSearch } from "../../utils/url";
+
+interface PathParams {
+  name: string;
+  scope?: string;
+  version: string;
+}
+
+export function Package() {
+  const { name, scope, version }: PathParams = useParams();
+
+  const [requestMetadata, metadataResponse] = useRequest<Metadata>(() =>
```

Comment:
I believe TS should be able to use type inference seeing that the return type of `fetchMetadata` is `Promise<MetaData>` and therefore it shouldn't be necessary for us to pass the type param here `useRequest(() => fetchMetaData(name, version. scope))` should work.

Reply:
I've actually re-written this a bit so that the inference is much better